platform/mellanox/mlnx-platform-api/sonic_platform/dpuctl_hwm.py

```diff
@@ -0,0 +1,236 @@
+#
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Class Implementation for per DPU functionality"""
+import os.path
+import time
+
+try:
+    from .inotify_helper import InotifyHelper
+    from sonic_py_common.logger import Logger
+    from sonic_platform import utils
+except ImportError as e:
+    raise ImportError(str(e)) from e
+
+EVENT_BASE = "/var/run/hw-management/events/"
+SYSTEM_BASE = "/var/run/hw-management/system/"
+CONFIG_BASE = "/var/run/hw-management/config"
+logger = Logger()
+
+WAIT_FOR_SHTDN = 60
+WAIT_FOR_DPU_READY = 60
+
+
+class DataWriter():
+    """Class for writing data to files"""
+    def __init__(self, file_name):
+        self.file_name = file_name
+        self.file_obj = None
+        if not os.path.isfile(self.file_name):
+            raise FileNotFoundError(f"File {self.file_name} does not exist!")
+
+    def __enter__(self):
+        self.file_obj = open(self.file_name, 'w', encoding="utf-8")
+        return self.file_obj
+
+    def __exit__(self, *args):
+        self.file_obj.close()
+
+
+class DpuCtlPlat():
+    """Class for Per DPU API Call"""
+    def __init__(self, dpu_index):
+        self.index = dpu_index + 1
+        self._name = f"dpu{self.index}"
+        self.set_go_down_path = os.path.join(SYSTEM_BASE,
+                                             f"dpu{self.index}_rst")
+        self.set_pwr_path = os.path.join(SYSTEM_BASE,
+                                         f"dpu{self.index}_pwr")
+        self.set_pwr_f_path = os.path.join(SYSTEM_BASE,
+                                           f"dpu{self.index}_pwr_force")
+        self.get_dpu_rdy_path = os.path.join(EVENT_BASE,
+                                             f"dpu{self.index}_ready")
+        self.set_dpu_perst_en_path = os.path.join(SYSTEM_BASE,
+                                                  f"dpu{self.index}_perst_en")
+
+    def write_file(self, file_name, content_towrite):
+        """Write given value to file only if file exists"""
+        try:
+            with DataWriter(file_name) as file_obj:
+                file_obj.write(content_towrite)
+        except (ValueError,
```

Comment:
except Exception is enough. You don't need to list all possible types

Reply:
Done